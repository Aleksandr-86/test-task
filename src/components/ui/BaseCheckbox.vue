<script setup lang="ts">
import { computed } from 'vue'

const props = defineProps<{
  modelValue: boolean
  value: boolean
  label: string
}>()

const emit = defineEmits(['update:modelValue'])

const model = computed({
  get() {
    return props.modelValue
  },
  set(value) {
    emit('update:modelValue', value)
  }
})
</script>

<template>
  <div class="flex items-center space-x-2 pl-1">
    <input
      class="form-checkbox h-5 w-5 cursor-pointer rounded outline-blue-700 checked:outline-blue-700 focus:outline-blue-700"
      type="checkbox"
      name="checkbox"
      v-model="model"
      :value="value" />
    <span class="text-lg font-bold text-gray-800 dark:text-gray-200">{{
      label
    }}</span>
  </div>
</template>
