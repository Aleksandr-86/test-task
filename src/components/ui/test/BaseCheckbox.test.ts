import { render } from '@testing-library/vue'
import BaseCheckbox from '@/components/ui/BaseCheckbox.vue'

describe('Компонент BaseCheckbox', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(BaseCheckbox, {
      props: {
        label: 'Кнопка',
        iconSrc: 'icons/key'
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
