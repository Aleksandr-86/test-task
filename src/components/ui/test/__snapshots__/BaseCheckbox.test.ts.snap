// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент BaseCheckbox > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="flex items-center space-x-2 pl-1"
        iconsrc="icons/key"
      >
        <input
          class="form-checkbox h-5 w-5 cursor-pointer rounded outline-blue-700 checked:outline-blue-700 focus:outline-blue-700"
          name="checkbox"
          type="checkbox"
          value="false"
        />
        <span
          class="text-lg font-bold text-gray-800 dark:text-gray-200"
        >
          Кнопка
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center space-x-2 pl-1"
      iconsrc="icons/key"
    >
      <input
        class="form-checkbox h-5 w-5 cursor-pointer rounded outline-blue-700 checked:outline-blue-700 focus:outline-blue-700"
        name="checkbox"
        type="checkbox"
        value="false"
      />
      <span
        class="text-lg font-bold text-gray-800 dark:text-gray-200"
      >
        Кнопка
      </span>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
